patches/server/0935-Add-LocalCoordinates-API.patch

```diff
@@ -0,0 +1,41 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: ChristopheG <61288881+chrisgdt@users.noreply.github.com>
+Date: Sat, 13 Aug 2022 13:57:26 +0200
+Subject: [PATCH] Add LocalCoordinates API
+
+
+diff --git a/src/main/java/net/minecraft/commands/arguments/coordinates/LocalCoordinates.java b/src/main/java/net/minecraft/commands/arguments/coordinates/LocalCoordinates.java
+index ba10bf752687ef49b16dea53c791d4dd7480cda2..6e89ef2f3db5d9c22bb1fc939b3a046c4e11c12f 100644
+--- a/src/main/java/net/minecraft/commands/arguments/coordinates/LocalCoordinates.java
++++ b/src/main/java/net/minecraft/commands/arguments/coordinates/LocalCoordinates.java
+@@ -24,6 +24,12 @@ public class LocalCoordinates implements Coordinates {
+     public Vec3 getPosition(CommandSourceStack source) {
+         Vec2 vec2 = source.getRotation();
+         Vec3 vec3 = source.getAnchor().apply(source);
++        // Paper start
++        return getPosition(vec3, vec2);
++    }
++
++    public Vec3 getPosition(Vec3 vec3, Vec2 vec2) {
```

Comment:
Maybe this should be a static method also taking the x y and z (left, up, forwards).
That way we don't have to allocate another throwaway object when calling this.